framework/contracts/native/ibc-host/src/endpoints/packet.rs

```diff
@@ -145,18 +148,39 @@ pub fn handle_host_module_action(
         _ => {}
     }
 
-    // We pass the message on to the module
-    let msg = wasm_execute(
-        target_module_resolved.address,
-        &MiddlewareExecMsg::ModuleIbc::<Empty, Empty>(ModuleIbcMsg {
-            client_chain,
-            source_module: source_module.module_info,
-            msg,
-        }),
-        vec![],
-    )?;
+    let response = Response::new().add_attribute("action", "module-ibc-call");
+    match msg {
+        ModuleActionMsg::Execute(msg) => {
+            // We pass the message on to the module
+            let msg = wasm_execute(
+                target_module_resolved.address,
+                &MiddlewareExecMsg::ModuleIbc::<Empty, Empty>(ModuleIbcMsg {
+                    client_chain,
+                    source_module: source_module.module_info,
+                    msg,
+                }),
+                vec![],
+            )?;
+
+            Ok(response.add_message(msg))
+        }
+        ModuleActionMsg::Query(msg) => {
+            // We query module and set data to the response that can be parsed later
+            let query = QueryRequest::<Empty>::from(WasmQuery::Smart {
+                contract_addr: target_module_resolved.address.into_string(),
+                msg,
+            });
+            let bin = match deps.querier.raw_query(&to_json_vec(&query)?) {
+                SystemResult::Err(system_err) => Err(StdError::generic_err(format!(
+                    "Querier system error: {system_err}"
+                ))),
+                SystemResult::Ok(ContractResult::Err(contract_err)) => Err(StdError::generic_err(
+                    format!("Querier contract error: {contract_err}"),
+                )),
+                SystemResult::Ok(ContractResult::Ok(value)) => Ok(value),
+            }?;
 
-    Ok(Response::new()
-        .add_attribute("action", "module-ibc-call")
-        .add_message(msg))
+            Ok(response.set_data(bin))
```

Comment:
I really don't like the fact that we use the execute data field to get a query result back. Do you have a reasoning behind not using actual queries to query data here and using the data field instead ?

Reply:
Otherwise, nice implementation

Reply:
> I really don't like the fact that we use the execute data field to get a query result back. Do you have a reasoning behind not using actual queries to query data here and using the data field instead ?

I don't really get what you mean by "using actual queries" here. Can you show what you mean? I don't know how to make polytone query desired module without calling execute on `ibc-host`

Reply:
And this is not compatible with https://github.com/AbstractSDK/abstract/pull/369 (adding multiple IBC queries simultaneously)

Reply:
> I don't really get what you mean by "using actual queries" here. Can you show what you mean? I don't know how to make polytone query desired module without calling execute on `ibc-host`

You can make polytone call the `query` on `ibc-host` instead of `execute

Reply:
Okay I think I've got what you mean! Create new query method on ibc-host and query from there. I just didn't think of that, lol

Reply:
Ahhh yeah that's a good suggestion!!